### ISSUE TITLE
Add at least one space around binary operators

### ISSUE THREAD

**Issue by duboism** [PHP-CS-Fixer/PHP-CS-Fixer#6861]
## Rule request

The proposal is to add an option to the `binary_operator_spaces` (say `at_least_single_space`) to add at least one space around binary operators (i.e. add a space before and after if there is none and leave things unchanged otherwise) and to include it by default in the `@PSR12` rule set.

The need for this rule was discussed in #6849: basically some old code doesn't respect PSR12 requirement to have at least one space around all binary operators so this rule allows to fix that (without breaking more complex PSR12-compliant patterns); including it in  `@PSR12` rule set allows to fix the old code base easily.

Also, I think that `PHP_CodeSniffer` does that when using the `PSR12` standard.

### Code snippet that shows how the rule would fix PHP code

Considering the snippet:
```php
$a =40; // To be fixed
$a+=1;  // To be fixed
$a = $a + 1; // No fix
$foo = 1; // No fix
$b   = 2; // No fix
```

One expect the following PSR12-compliant fix:
```php
$a = 40; // To be fixed
$a += 1;  // To be fixed
$a = $a + 1; // No fix
$foo = 1; // No fix
$b   = 2; // No fix
```

**Comment by duboism** [comment]
As mentioned in #6849, I'm unable to implement this but I can help testing it.

**Comment by duboism** [comment]
Hello,

I have noticed that PR #6878 has been closed without being merged. Any plan to work on that ?

**Comment by Wirone** [comment]
@duboism I really doubt that any of core team members will work on this, there is too much more important things to do right now.

**Comment by SpacePossum** [comment]
not sure I follow, you can either set the fixing to have a space before and after, or align (so add space before) and have one after, if this is both not wanted than not use the rule?

**Comment by duboism** [comment]
> not sure I follow, you can either set the fixing to have a space before and after, or align (so add space before) and have one after, if this is both not wanted than not use the rule?

The motivation behind the rule was that I work on an poorly formatted code base. In particular, it contains a lot of assignments of the form `$b=1`. The current `@PSR12` ruleset (at the time of discussion #6849) doesn't include any rule regarding spaces around binary operators so it won't change those (thus not respecting PSR12).

Our first suggestion was to make `single_space` the default but @kubawerlos objected that this would beak other legitimate formatting (aligned assignments for example) and he suggested `at_least_single_space` as a tradeoff (IMHO this should be included in `@PSR12`).

**Comment by duboism** [comment]
Since #6878 is closed I opened a new PR (#7175) for that.

**Comment by duboism** [comment]
Now that #7175 is merged, I think that we can close this issue.
